pkg/reduce/metrics.go

```diff
@@ -0,0 +1,28 @@
+package reduce
+
+import (
+	metricspkg "github.com/numaproj/numaflow/pkg/metrics"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+// readMessagesCount is used to indicate the number of messages read
+var readMessagesCount = promauto.NewCounterVec(prometheus.CounterOpts{
+	Subsystem: "isb_reader",
```

Comment:
What is our strategy here for the `Subsystem`?

In isb forward (map), it's `forwarder`, in jetstream, it's `isb_jetstream`, here it's `isb_reader`. Shall we use a explicit name like `reduce_isb_reader` for it?